config/confighttp/confighttp.go

```diff
@@ -119,6 +120,21 @@ func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Ext
 	}, nil
 }
 
+func (hcs *HTTPClientSettings) WithDefaultUserAgent(defaultUserAgent string) {
+	userAgentFound := false
+	for k := range hcs.Headers {
+		if strings.ToLower(k) == "user-agent" {
+			userAgentFound = true
+		}
+	}
+	if !userAgentFound {
+		if hcs.Headers == nil {
+			hcs.Headers = map[string]string{}
+		}
+		hcs.Headers["User-Agent"] = defaultUserAgent
```

Comment:
I would say that this is not a "clean" pattern, we should not change the user's config if possible.

Reply:
Hmm, I see what you're saying. How about adding the default user-agent as part of the default client settings? https://github.com/open-telemetry/opentelemetry-collector/blob/1d4b414590f9b41377df560a452638cbf6358732/exporter/otlphttpexporter/factory.go#L45

Reply:
(assuming the build info is available at that point in the lifecycle)

Reply:
+1 to what @bogdandrutu said. 

The reason for that is because we want to be able to show user-defined config exactly as it is (e.g. in zpages) and modifying it here will set a bad precedent that what we show is not exactly what the user defined, but may be different from it. I think it is undesirable.

Perhaps a better place to inject the default value for "User-Agent" header is in ToClient() func. Check that it is not specified by the user and add the default.

Reply:
> I would say that this is not a "clean" pattern, we should not change the user's config if possible.

I don't see this as changing the user's config any more than it would be if we did this as an option on `ToClient`.  If the user provides a user agent header, it is used, otherwise a default specified by the component using the HTTP client is used.  As for zpages, it might be odd that it wouldn't reflect this configuration until first use, but I think having it reflect the actual configuration is useful.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/pull/3970#issuecomment-936850553